app/addons/documents/header/header.react.jsx

```diff
@@ -0,0 +1,239 @@
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License. You may obtain a copy of
+// the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+
+define([
+  'app',
+  'api',
+  'react',
+  'addons/documents/header/header.stores',
+  'addons/documents/header/header.actions'
+],
+
+function (app, FauxtonAPI, React, Stores, Actions) {
+  var ReactCSSTransitionGroup = React.addons.CSSTransitionGroup;
+  var headerBarStore = Stores.headerBarStore;
+  var alternativeHeaderBarStore = Stores.alternativeHeaderBarStore;
+
+  // this will be a global component
+  var ToggleHeaderButton = React.createClass({
+    render: function () {
+      var iconClasses = 'icon ' + this.props.fonticon + ' ' + this.props.innerClasses,
+          containerClasses = 'button ' + this.props.containerClasses;
+
+      if (this.props.setEnabledClass) {
+        containerClasses = containerClasses + ' js-headerbar-togglebutton-selected';
+      }
+
+      return (
+        <button
+          title={this.props.title}
+          disabled={this.props.disabled}
+          onClick={this.props.toggleCallback}
+          className={containerClasses}
+          >
+          <i className={iconClasses}></i><span>{this.props.text}</span>
+        </button>
+      );
+    }
+  });
+
+  var AlternateHeaderControlBarController = React.createClass({
```

Comment:
I think we should rename the `AlternativeHeader` to something like `BulkDoumentHeader`. Since when that header is displayed we editing the documents in bulk. Its a little more descriptive than just `AlternativeHeader`